### PR TITLE
A few features and bugfixes

### PR DESCRIPTION
**Pull request by tweirtx**

### CODE REVIEW

File: dozer/cogs/info.py [hunk 1]
```diff
@@ -10,48 +12,85 @@
...
-        """
-        Retrieve information about a member of the guild.
-        If no arguments are passed, information about the author is used.
-        **This command works without mentions.** Remove the '@' before your mention so you don't ping the person unnecessarily.
-        You can pick a member by:
-        - Username (`cooldude`)
-        - Username and discriminator (`cooldude#1234`)
-        - ID (`326749693969301506`)
-        - Nickname - must be exact and is case-sensitive (`"Mr. Cool Dude III | Team 1234"`)
-        - Mention (not recommended) (`@Mr Cool Dude III | Team 1234`)
```
Comment (octocynth) [inline dozer/cogs/info.py:27]: This information seems reasonable to keep as-is or move to the example usage. Anecdotally, `&user` seems to be used most often with mentions, and pingsock/angryping type responses are fairly frequent.

File: dozer/cogs/info.py [hunk 2]
```diff
@@ -10,48 +12,85 @@
...
+
+        status = 'DND' if member.status is discord.Status.dnd else member.status.name.title()
+        if member.status is not discord.Status.offline:
+            platforms = self.pluralize([platform for platform in ('web', 'desktop', 'mobile') if
+                                        getattr(member, f'{platform}_status') is not discord.Status.offline])
+            status = f'{status} on {platforms}'
+        activities = '\n'.join(self._format_activities(member.activities))
+        embed.add_field(name='Status and Activity', value=f'{status}\n{activities}', inline=True)
+
+        embed.add_field(name='Roles', value=', '.join(role.name for role in member.roles[:0:-1]) or 'None')
```
Comment (octocynth) [inline dozer/cogs/info.py:44]: This field overflowing is something we might want to prepare for in the future. For now, the one closest to overflowing is a test account with every role, and that only reaches 900, so this PR isn't blocked on handling >1024 characters correctly.

File: dozer/cogs/info.py [hunk 3]
```diff
@@ -10,48 +12,85 @@
...
+        embed.set_thumbnail(url=icon_url)
+        await ctx.send(embed=embed)
 
     member.example_usage = """
-    `{prefix}member` - get information about yourself
-    `{prefix}member cooldude#1234` - get information about cooldude
-    """
+        `{prefix}member`: show your member info
+        `{prefix}member {ctx.me}`: show my member info
+        """
```
Comment (octocynth) [inline dozer/cogs/info.py:52]: This indentation is inconsistent with other example usages.

File: dozer/cogs/info.py [hunk 4]
```diff
@@ -10,48 +12,85 @@
...
+                matcher.set_seq1(filtered_activity.name)
+                if matcher.quick_ratio() < 0.6 and matcher.ratio() < 0.6:  # Use quick_ratio if we can as ratio is slow
+                    filtered.append(activity)
+                    break
+
+        return [format_activity(activity) for activity in filtered]
+
+    @staticmethod
+    def pluralize(values: typing.List[str]) -> str:
+        """Pluralizes things"""
```
Comment (octocynth) [inline dozer/cogs/info.py:85]: This doc comment isn't very informative to a reader. It would be helpful to explain what the function does in more detail.

File: dozer/cogs/info.py [hunk 5]
```diff
@@ -66,11 +105,16 @@ async def guild(self, ctx):
         e.add_field(name='Created at', value=guild.created_at.strftime(datetime_format))
         e.add_field(name='Owner', value=guild.owner)
         e.add_field(name='Members', value=guild.member_count)
-        e.add_field(name='Channels', value=len(guild.channels))
-        e.add_field(name='Roles', value=len(guild.roles) - 1)  # Remove @everyone
-        e.add_field(name='Emoji', value=len(guild.emojis))
+        e.add_field(name='Channels', value=str(len(guild.channels)))
+        e.add_field(name='Roles', value=str(len(guild.roles) - 1))  # Remove @everyone
+        e.add_field(name='Emoji', value=str(len(guild.emojis)))
```
Comment (octocynth) [inline dozer/cogs/info.py:110]: It would be useful to count static vs animated emojis and show those totals here as well.

File: dozer/cogs/teams.py [hunk 6]
```diff
@@ -73,10 +73,17 @@ async def onteam(self, ctx, team_type, team_number: int):
             e = discord.Embed(type='rich')
             e.title = 'Users on team {}'.format(team_number)
             e.description = "Users: \n"
+            extra_mems = ""
             for i in users:
                 user = ctx.guild.get_member(i.user_id)
                 if user is not None:
-                    e.description = "{}{} {} \n".format(e.description, user.display_name, user.mention)
+                    memstr = "{} {} \n".format(user.display_name, user.mention)
+                    if len(e.description + memstr) > 2047:
```
Comment (octocynth) [inline dozer/cogs/teams.py:81]: What is the 2047 magic number here?

Comment (tweirtx) [reply dozer/cogs/teams.py:81]: The maximum length of a description

File: dozer/cogs/info.py (hunk 5 above)
Comment (tweirtx) [reply dozer/cogs/info.py:110]: Maybe, I'll go steal more code

File: dozer/cogs/info.py (hunk 4 above)
Comment (tweirtx) [reply dozer/cogs/info.py:85]: I honestly don't know what it does, I stole it from you

File: dozer/cogs/info.py (hunk 3 above)
Comment (tweirtx) [reply dozer/cogs/info.py:52]: Oops

File: dozer/cogs/info.py (hunk 2 above)
Comment (tweirtx) [reply dozer/cogs/info.py:44]: This is code I got from you :stuck_out_tongue:

File: dozer/cogs/info.py (hunk 1 above)
Comment (tweirtx) [reply dozer/cogs/info.py:27]: Ok

File: dozer/cogs/info.py (hunk 2 above)
Comment (octocynth) [reply dozer/cogs/info.py:44]: I'm no less fallible than anyone else, and I critique my own code as hard if not harder than anyone else's. 😉 But since this is a non-issue, I'm marking it as resolved.

File: dozer/cogs/info.py (hunk 4 above)
Comment (octocynth) [reply dozer/cogs/info.py:85]: Given a list of items, it inserts commas and "and"s in the right places to create a grammatically correct list.

Comment (tweirtx) [reply dozer/cogs/info.py:85]: Ok thanks

File: dozer/cogs/info.py [hunk 7]
```diff
@@ -10,67 +12,119 @@
...
+                matcher.set_seq1(filtered_activity.name)
+                if matcher.quick_ratio() < 0.6 and matcher.ratio() < 0.6:  # Use quick_ratio if we can as ratio is slow
+                    filtered.append(activity)
+                    break
+
+        return [format_activity(activity) for activity in filtered]
+
+    @staticmethod
+    def pluralize(values: typing.List[str]) -> str:
+        """Given a list of items, it inserts commas and "and"s in the right places to create a grammatically correct list."""
```
Comment (octocynth) [inline dozer/cogs/info.py:93]: Copying and pasting my comment into a docstring isn't what I meant. 😂 Please rephrase this to fill in the blank in "What does this function do? It ___" as most of our other function docs do.
